src/rt/minfo.d

```diff
@@ -505,12 +513,11 @@ struct ModuleGroup
         }
 
         // finally, do the sorting for both shared and tls ctors.
-        try
-        {
-            _ctors = doSort(MIctor | MIdtor);
-            _tlsctors = doSort(MItlsctor | MItlsdtor);
-        }
-        catch(DeprecatedCycleException)
+        bool cycle = false;
+        _ctors = doSort(MIctor | MIdtor, cycle);
```

Comment:
I implemented the suggested behavior, but changed the return to a ref parameter which simplifies the logic.

In general, the GC should not be used in druntime if it can be avoided.

Exceptions are expensive (both to throw and in code size) and should be avoided where reasonable.